File: app/src/main/java/com/sd/lib/context/demo/MainActivity.java
package com.sd.lib.context.demo;

import android.os.Bundle;
import android.util.Log;

import androidx.appcompat.app.AppCompatActivity;

import com.sd.lib.context.FContext;

public class MainActivity extends AppCompatActivity {
    public static final String TAG = MainActivity.class.getSimpleName();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Log.i(MainActivity.TAG, "onCreate:" + this + " context:" + FContext.get());
    }
}

File: app/src/main/java/com/sd/lib/context/demo/MainActivity.kt
package com.sd.lib.context.demo

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import com.sd.demo.context.R
import com.sd.lib.context.FContext

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        logMsg { "Activity onCreate context:${FContext.get()}" }
    }
}

inline fun logMsg(block: () -> String) {
    Log.i("context-demo", block())
}